### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by aljones15**
this removes all lint errors in all files while also enabling jsdoc lint rules.
I would use that project that auto-generates the markdown files form jsdoc, but it's not working for some reason.

### CODE REVIEW

File: ControllerKey.js [hunk 1]
```diff
@@ -116,18 +116,19 @@ export class ControllerKey {
    * wrap or unwrap keys.
    *
    * If this ControllerKey is a controller of the KEK, then the API for it can
-   * returned by passing only the key description. Otherwise, an OCAP-LD
-   * authorization capability must also be passed; without this capability,
-   * calls to the returned API will not be authorized to perform KEK operations.
+   * be returned by passing only the key id and type.
+   * Otherwise, an OCAP-LD authorization capability must also be passed;
```
Comment (dlongley) [inline ControllerKey.js]: Either add whitespace above this or merge into the above paragraph.

File: ControllerKey.js [hunk 2]
```diff
@@ -19,30 +19,30 @@ const _seedCache = new SeedCache();
...
-   * instance.
-   *
+   * Creates a new instance of a ControllerKey.
    * A ControllerKey can be passed as an `invocationSigner` to a KmsClient, but
    * a KmsClient instance is typically used internally by other instances that
    * can be created via the ControllerKey API such as instances of the Kek
    * and Hmac classes.
    *
+   * This function should never be called directly.
+   * Use one of these methods to create a ControllerKey instance.
```
Comment (dlongley) [inline ControllerKey.js]: Merge into a single paragraph or add whitespace to separate paragraphs here and above.

Comment (dlongley) [reply ControllerKey.js]: The rule is: Paragraphs should have an empty whitespace line after them (even if they are a single sentence).

File: ControllerKey.js [hunk 3]
```diff
@@ -141,18 +142,19 @@ export class ControllerKey {
    * sign or verify data.
    *
    * If this ControllerKey is a controller of the HMAC, then the API for it can
-   * returned by passing only the key description. Otherwise, an OCAP-LD
-   * authorization capability must also be passed; without this capability,
+   * be returned by passing only the key id and type.
+   * Otherwise, an OCAP-LD authorization capability must also be passed;
+   * without this capability,
```
Comment (dlongley) [inline ControllerKey.js]: Same whitespace issues here, apply above rule.

File: ControllerKey.js [hunk 4]
```diff
@@ -166,22 +168,22 @@ export class ControllerKey {
...
    *
    * If this ControllerKey is a controller of the AsymmetricKey, then the API
-   * for it can returned by passing only the key description. Otherwise, an
-   * authorization capability must also be passed; without this capability,
-   * calls to the returned API will not be authorized to perform asymmetric key
-   * operations.
+   * for it can be returned by passing only the id and type.
+   * Otherwise, an authorization capability must also be passed;
+   * without this capability, calls to the returned API will not be
+   * authorized to perform asymmetric key operations.
```
Comment (dlongley) [inline ControllerKey.js]: Same problem here.

File: ControllerKey.js [hunk 5]
```diff
@@ -307,7 +309,7 @@ export class ControllerKey {
    * via `fromSecret` with `cache` set to `true` in order to ensure the key
    * cannot be loaded via `fromCache`.
    *
-   * @param {Object} options - The options to use.
+   * @param {object} options - The options for the caches.
```
Comment (dlongley) [inline ControllerKey.js]: I don't understand "for the caches"... seems the previous was better even if vague. Did you mean "for the cache"? Even then I'm not sure what that means. Also, everything else is "The options to use." so let's keep it the same for easier maintenance.

File: ControllerKey.js (hunk 1 above)
Comment (aljones15) [reply ControllerKey.js]: Merged into a single paragraph here: https://github.com/digitalbazaar/web-kms-client/pull/2/commits/26ca0d200d9e088e209e8a2dd1982b1f4de56bec

File: ControllerKey.js (hunk 3 above)
Comment (aljones15) [reply ControllerKey.js]: Merged into one paragraph: https://github.com/digitalbazaar/web-kms-client/pull/2/commits/26ca0d200d9e088e209e8a2dd1982b1f4de56bec

File: ControllerKey.js (hunk 4 above)
Comment (aljones15) [reply ControllerKey.js]: Merged into one paragraph here: https://github.com/digitalbazaar/web-kms-client/pull/2/commits/26ca0d200d9e088e209e8a2dd1982b1f4de56bec

File: ControllerKey.js (hunk 5 above)
Comment (aljones15) [reply ControllerKey.js]: This was reverted to `The options to use.` here: https://github.com/digitalbazaar/web-kms-client/pull/2/commits/48dd98df86624e1849176410812202c047112ec3

File: ControllerKey.js (hunk 2 above)
Comment (aljones15) [reply ControllerKey.js]: Addressed here, but I added one line of test and made it a single paragraph: https://github.com/digitalbazaar/web-kms-client/pull/2/commits/ba4f52d638724b1761a012e4a1290affd65df190

File: ControllerKey.js [hunk 6]
```diff
@@ -116,18 +118,19 @@ export class ControllerKey {
    * wrap or unwrap keys.
    *
    * If this ControllerKey is a controller of the KEK, then the API for it can
-   * returned by passing only the key description. Otherwise, an OCAP-LD
-   * authorization capability must also be passed; without this capability,
-   * calls to the returned API will not be authorized to perform KEK operations.
+   * be returned by passing only the key id and type. Otherwise,
+   * an OCAP-LD authorization capability must also be passed;
+   * without this capability, calls to the returned API will not be authorized
+   * to perform KEK operations.
```
Comment (dlongley) [inline ControllerKey.js]: The above is still very sub-optimally spaced :( ... doesn't have to be perfect, but looks pretty goofy right now.

File: ControllerKey.js [hunk 7]
```diff
@@ -141,18 +144,18 @@ export class ControllerKey {
...
    *
    * If this ControllerKey is a controller of the HMAC, then the API for it can
-   * returned by passing only the key description. Otherwise, an OCAP-LD
-   * authorization capability must also be passed; without this capability,
-   * calls to the returned API will not be authorized to perform HMAC
-   * operations.
+   * be returned by passing only the key id and type. Otherwise,
+   * an OCAP-LD authorization capability must also be passed;
+   * without this capability, calls to the returned API will not
+   * be authorized to perform HMAC operations.
```
Comment (dlongley) [inline ControllerKey.js]: Same thing with clearly sub-optimal spacing, sorry to be a drag on this.

File: Kek.js [hunk 8]
```diff
@@ -12,13 +12,14 @@ const JOSE_ALGORITHM_MAP = {
 export class Kek {
   /**
    * Creates a new instance of a key encryption key.
+   * Used to protect the master encryption key e.g. Controller Key.
```
Comment (dlongley) [inline Kek.js]: The `ControllerKey` is not a content encryption key, suggest:

```suggestion
   * Used to protect the content encryption key.
```

Also ... please fix the whitespace issue here like the others.

File: KmsClient.js [hunk 9]
```diff
@@ -10,13 +10,21 @@ import {signCapabilityInvocation} from 'http-signature-zcap-invoke';
 const SECURITY_CONTEXT_V2_URL = 'https://w3id.org/security/v2';
 const DEFAULT_HEADERS = {Accept: 'application/ld+json, application/json'};
 
+/**
+ * A node https module.
+ *
+ * @typedef {object} https
+ * @see https://nodejs.org/docs/latest-v11.x/api/https.html
```
Comment (dlongley) [inline KmsClient.js]: I don't understand this... is this a copy+paste error?

File: KmsClient.js [hunk 10]
```diff
@@ -10,13 +10,21 @@ import {signCapabilityInvocation} from 'http-signature-zcap-invoke';
...
   /**
    * Creates a new KmsClient.
    *
-   * @param {Object} options - The options to use.
-   * @param {string} [keystore=undefined] the ID of the keystore that must be a
-   *   URL that refers to the keystore's root storage location; if not given,
+   * @param {object} options - The options to use.
+   * @param {string} [options.keystore=undefined] - The ID of the keystore
+   * that must be a URL that refers to the keystore's root storage
+   * location; if not given,
```
Comment (dlongley) [inline KmsClient.js]: Fix spacing (indent 2).

Comment (aljones15) [reply KmsClient.js]: Addressed in https://github.com/digitalbazaar/web-kms-client/pull/2/commits/09d4ec93574132f2a7a5205559256f63a81ca903

File: KmsClient.js (hunk 9 above)
Comment (aljones15) [reply KmsClient.js]: Remove it here: https://github.com/digitalbazaar/web-kms-client/pull/2/commits/09d4ec93574132f2a7a5205559256f63a81ca903

File: ControllerKey.js (hunk 7 above)
Comment (aljones15) [reply ControllerKey.js]: Addressed in https://github.com/digitalbazaar/web-kms-client/pull/2/commits/09d4ec93574132f2a7a5205559256f63a81ca903

File: ControllerKey.js (hunk 6 above)
Comment (aljones15) [reply ControllerKey.js]: Addressed in https://github.com/digitalbazaar/web-kms-client/pull/2/commits/09d4ec93574132f2a7a5205559256f63a81ca903

**Pull request merged**